package NSnippets.NProto;

option optimize_for = SPEED;

import "library/cpp/langmask/proto/langmask.proto";

/*
So here we are.
When modifying this file make sure all its versions are compatible (at least stable/pre-stable/trunk ones).
- no required fields
- no field type changes
- do not reuse old field numbers
- keep field numbers and types same across versions
- avoid backporting new fields, just add the field in trunk first
- once again, when porting new field, keep teh number; gaps are ok
*/

message TSnipInfoReqParams {
    optional uint32 InfoRequestType = 1;
    optional uint32 PageNumber = 2;
    optional bool UnpackedOnly = 3;
    optional bool ShowMarkup = 4;
    optional string AlgoName = 5;
    optional bool IsLinkArchive = 6;
    optional bool IsJson = 7;
}

message TNamedValue {
    optional string Name = 1;
    optional string Value = 2;
}

message TSnipReqParams {
    optional bool PornoFactor = 1;
    optional bool ImagesPornoFactor = 2;
    optional int32 ExtendSnippets = 3;
    optional int32 MaxTitleLength = 4;
    optional uint64 ImagesQueryLang = 7;
    optional int32 MaxSnippetCount = 8;
    optional bool AllPassageHits = 9;
    optional bool MarkParagraphs = 10;
    optional bool AllowYacaSnippets = 11;
    optional int32 RequestedSnippetLength = 12;
    optional bool IsMobileSearch = 13;
    optional uint64 FaceType = 14;
    optional bool RawPassages = 15;
    optional bool DumpCandidates = 16;
    optional bool DumpAllAlgo3Pairs = 17;
    optional bool SlovariSearch = 18;
    optional bool ImgsearchSnippets = 19; // just like old BuildForImg
    optional int32 MaxTitleStartLen = 20;
    optional int32 MaxTitlePixelLen = 21;
    optional bool MainPage = 22;
    optional TSnipInfoReqParams InfoRequestParams = 23;
    optional bytes SnippetExperiments = 24;
    optional bytes ForeignNavHackLang = 25;
    optional bool IsTelOrg = 26;
    optional uint64 TitleCutMethod = 27;
    optional bool DumpWithUnusedFactors = 28;
    optional bool DumpWithManualFactors = 29;
    optional uint64 QueryLangMaskLow = 30; // obsolete, use QueryLangMask instead
    optional int32 SignatureWidth = 31;
    optional int32 SignatureMinLength = 32;
    optional bool ShouldCalculateSignature = 33;
    optional bool IsPolitician = 34;
    optional bytes SiteAttr = 35;
    optional bool ShouldReturnDocStaticSig = 36;
    optional int64 DocStaticSig = 37;
    optional uint64 SnippetCutMethod = 43;
    optional float RequestedSnippetLengthRow = 44;
    optional int32 SnippetsAltHeader = 45;
    optional bytes UILang = 47;
    optional bool PaintedRawPassages = 48;
    optional bool AbsentT = 49;
    optional bool IsUrlOrEmailQuery = 50;
    optional bool VideoWatch = 52;
    optional int32 ImgFrameWidth = 53;
    optional int32 ImgFrameHeight = 54;
    optional int32 ImgPreviewThumbHeight = 55;
    optional int32 ScreenWidth = 56;
    optional int32 ScreenHeight = 57;
    optional int32 ImgFitSizeCount = 58;
    optional int32 ImgPreviewCount = 59;
    optional int32 ImgResizedCount = 60;
    optional int32 PornoFilterMode = 61;
    optional int32 PornoQueryType = 62;
    optional int32 ImgFSFrameWidth = 63;
    optional int32 ImgFSFrameHeight = 64;
    optional int32 ImgFSPreviewCount = 65;
    optional bytes ImgSeriesPage = 66;
    optional bool ForceImgSeries = 67;
    optional bool UseStaticDataOnly = 68;
    optional bytes StaticDataSource = 69;
    optional bool PaintNoAttrs = 70;
    optional bytes ImgFilter = 71;
    optional int32 ImgCBIRCount = 72;
    optional bool ImgDupsTableMode = 73;
    optional bool CPreview = 74;
    optional int32 CPreviewLength = 75;
    optional int32 ImgAllCount = 76;
    optional int32 CPreviewTitleLength = 77;
    optional bool IsTurkey = 78;
    optional bool UnpackFirst10 = 79;
    optional int32 ImgAllOffset = 80;
    optional int32 ImgCBIRSat = 81;
    optional bool VideosearchSnippets = 82; // like BuildForVideo
    optional uint64 RelevRegion = 83;
    optional bool ImgAllByRelev = 84; //not stable yet
    optional int32 ImgSizedCount = 85; //not stable yet
    optional uint64 ImgRegion = 86;
    optional bytes TLD = 87; //set on images serp at least
    optional int32 ImgUserWallpaperCount = 88;
    optional bytes EntityJson = 89;
    repeated bytes SnipParamsForDebug = 90;
    optional bool ImgPhoneWallpaperDups = 91;
    optional uint32 HitsTopLength = 92;
    optional bool ForcePureTitle = 93;
    optional float CommercialFactor = 94;
    optional bool IgnoreRegionPhrase = 95;
    optional uint32 MaxUrlLength = 96;
    optional uint32 UrlCutThreshold = 97;
    optional uint32 MaxUrlmenuLength = 99;
    optional bool ImgPatchThumbId = 100;
    optional int32 SnipWidth = 101;
    optional bool RTYSnippets = 102;
    optional bytes CompressedFactors = 103;
    optional uint64 UserReqHash = 104;
    optional .NProto.TLangMask QueryLangMask = 105;
    optional bytes RelevRegionName = 106;
    optional bool MainBna = 107;
    optional bytes Report = 108;
    optional bytes FactorsBorders = 109;
    optional uint32 PassageReplyFillerType = 110; // EPassageReplyFillerType
    repeated TNamedValue RelevParams = 111;
    repeated float UncompressedFactors = 112 [packed=true];
    optional bool ImgNeedBigThumbIds = 113;
}

message THitsCtx {
    optional uint64 Kind = 1;
    repeated uint32 TextHitsPlain = 2 [packed=true]; //for old contexts
    repeated uint32 LinkHitsPlain = 3 [packed=true];
    optional bool IsNav = 6;
    repeated uint32 MoreTextHitsPlain = 7 [packed=true]; //for even older contexts
    optional bool IsRoot = 9;
    optional int64 DocStaticSig = 10;
    optional bool WatchVideo = 11;
    optional bool IsTurkey = 12;
    repeated uint32 THSents = 13 [packed=true];
    repeated uint32 THMasks = 14 [packed=true];
    repeated float FormWeights = 15;
    optional bool IsFakeForBan = 16;
    optional bool IsFakeForRedirect = 17;
    optional bool IsSpok = 18;
}

message TArc {
    optional bool Valid = 1;
    optional bytes Data = 2;
    optional bytes ExtInfo = 3;
}

message TQuery {
    optional bytes QtreeBase64 = 1;
    optional bytes UserRequest = 2;
    optional bytes RegionPhraseTreeBase64 = 3;
    optional bytes MoreHlTreeBase64 = 4;
}

message TContextDownloadLog {
    optional string RequestId = 1;
    optional uint32 RegionId = 2;
    
    // Actual response from middlesearch
    optional string HeadlineSrc = 3;
    optional string Headline = 4;
    optional string SnippetGta = 5;
}

message TSnippetsCtx {
    optional TQuery Query = 1;
    optional TSnipReqParams ReqParams = 2;
    optional TArc TextArc = 3;
    optional TArc LinkArc = 4;
    optional THitsCtx Hits = 5;
    optional TContextDownloadLog Log = 6;
}

